docs/experiments/Data_Set.ipynb

```diff
@@ -0,0 +1,490 @@
+    "    test = Dataset(X_test, Y_test, y_test, int(n_classes))\n",
+    "    return training, test\n",
+    "\n",
+    "\n",
+    "def import_IB_data_from_mat(name_ID):\n",
+    "    \"\"\" Writes a .npy file to disk containing the harmonics dataset used by Tishby\n",
+    "    Args:\n",
+    "        name_ID: Identifier which is going to be part of the output filename\n",
+    "\n",
+    "    Returns:\n",
+    "        None\n",
+    "    \"\"\"\n",
+    "    print('Loading Data...')\n",
+    "    d = sio.loadmat('datasets/var_u.mat')\n",
+    "    F = d['F']\n",
+    "    y = d['y']\n",
+    "    C = type('type_C', (object,), {})\n",
+    "    data_sets_original = C()\n",
+    "    data_sets_original.data = F\n",
+    "    data_sets_original.labels = np.squeeze(np.concatenate((y[None, :], 1 - y[None, :]), axis=0).T)\n",
+    "\n",
+    "    data_sets = utils.data_shuffle(data_sets_original, 80, shuffle_data=True)\n",
+    "    X_train, y_train, X_test, y_test = data_sets.train.data, data_sets.train.labels[:,\n",
+    "                                                             0], data_sets.test.data, data_sets.test.labels[:, 0]\n",
+    "    np.savez_compressed('datasets/IB_data_' + str(name_ID), X_train=X_train, y_train=y_train, X_test=X_test, y_test=y_test)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Next, we load the data set."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_data, test_data = load()\n",
+    "X = np.concatenate([train_data[0], test_data[0]])\n",
+    "Y = np.concatenate([train_data[2], test_data[2]])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Then, we sum over the 12 binary inputs of $X$ and determine the share of $Y=1$ per each possible sum value. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Share of Y=1 per each value of the sum of the binary inputs of X:\n",
+      "0 :  0.0 %\n",
+      "1 :  0.0 %\n",
+      "2 :  0.0 %\n",
+      "3 :  0.0 %\n",
+      "4 :  0.0 %\n",
+      "5 :  7.575757575757576 %\n",
+      "6 :  57.57575757575758 %\n",
+      "7 :  92.42424242424242 %\n",
+      "8 :  100.0 %\n",
+      "9 :  100.0 %\n",
+      "10 :  100.0 %\n",
+      "11 :  100.0 %\n",
+      "12 :  100.0 %\n"
+     ]
+    }
+   ],
+   "source": [
+    "X_sum_Y = []\n",
+    "\n",
+    "for i in range(13):\n",
+    "    X_sum_Y.append([])\n",
+    "\n",
+    "for i in range(len(X)):\n",
+    "    j = np.sum(X[i])\n",
+    "    X_sum_Y[j].append(Y[i])\n",
+    "\n",
+    "print('Share of Y=1 per each value of the sum of the binary inputs of X:')\n",
+    "for i in range(13):\n",
+    "    print(i, ': ', 100 * np.sum(np.array(X_sum_Y[i]))/len(X_sum_Y[i]), '%')"
```

Comment:
we try to use f'' stings whenever possible, so this could be sth like:
    proportion = 100 * np.sum(np.array(X_sum_Y[i]))/len(X_sum_Y[i])
    print(f'{i}: {proportion}%')